security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/ExternalSecurityAnalysis.java

```diff
@@ -83,99 +83,16 @@ public boolean removeInterceptor(SecurityAnalysisInterceptor interceptor) {
-            super.after(workingDir, report);
-            LOGGER.debug("End of command execution in {}. ", workingDir);
-            return readResults(workingDir);
-        }
-
-        private void copyInputFiles(Path workingDir) throws IOException {
-            // copy input files to slave workingDir
-            Path networkDest = workingDir.resolve(NETWORK_FILE);
-            LOGGER.debug("Copying network to file {}", networkDest);
-            NetworkXml.write(network, networkDest);
-
-            Path dslFileDest = workingDir.resolve(CONTINGENCIES_FILE);
-            LOGGER.debug("Writing contingencies to file {}", dslFileDest);
-            Files.write(dslFileDest, contingenciesProvider.asScript().getBytes(StandardCharsets.UTF_8));
-
-            Path parametersFileDest = workingDir.resolve(PARAMETERS_FILE);
-            LOGGER.debug("Writing parameters to file {}", parametersFileDest);
-            JsonSecurityAnalysisParameters.write(parameters, parametersFileDest);
-        }
-
-        /**
-         * Handles case-file, contingencies-file, parameters-file, and extensions arguments.
-         */
-        protected SimpleCommandBuilder baseCommand(String id) {
-            SimpleCommandBuilder cmdBuilder = new SimpleCommandBuilder()
-                    .id(id)
-                    .program(config.getItoolsCommand())
-                    .arg("security-analysis")
-                    .option(CASE_FILE_OPTION, NETWORK_FILE)
-                    .option(CONTINGENCIES_FILE_OPTION, CONTINGENCIES_FILE)
-                    .option(PARAMETERS_FILE_OPTION, PARAMETERS_FILE);
-            if (!extensions.isEmpty()) {
-                cmdBuilder.option("with-extensions", String.join(",", extensions));
-            }
-            return cmdBuilder;
-        }
-
-        protected List<CommandExecution> buildCommandExecution() {
-            SimpleCommandBuilder builder = baseCommand("security-analysis")
-                    .option(OUTPUT_FILE_OPTION, "JSON")
-                    .option(OUTPUT_FILE_OPTION, OUTPUT_FILE);
-            if (taskCount != null) {
-                builder.option(TASK_COUNT, Integer.toString(taskCount));
-            }
-            SimpleCommand cmd = builder.build();
-            return Collections.singletonList(new CommandExecution(cmd, 1, 1));
-        }
-
-        /**
-         * Read results from the working dir: may be overriden by child implementations.
-         */
-        protected SecurityAnalysisResult readResults(Path workingDir) {
-            Path taskResultFile = workingDir.resolve(OUTPUT_FILE);
-            return SecurityAnalysisResultDeserializer.read(taskResultFile);
-        }
+    protected SecurityAnalysisExecutionInput buildInput(String workingStateId, SecurityAnalysisParameters parameters, ContingenciesProvider contingenciesProvider) {
+        SecurityAnalysisExecutionInput input = new SecurityAnalysisExecutionInput();
```

Comment:
Could you make the API more fluent? Maybe you already do it, but you don’t use it